langchain/src/vectorstores/mongodb_atlas.ts

```diff
@@ -171,13 +171,17 @@ export class MongoDBAtlasVectorSearch extends VectorStore {
     );
 
     return mmrIndexes.map((idx) => {
-      const doc = resultDocs[idx][0];
+      const [doc, score] = resultDocs[idx];
 
       // remove embeddings if they were not requested originally
       if (!includeEmbeddingsFlag) {
         delete doc.metadata[this.embeddingKey];
       }
-      return doc;
+      return {
+        pageContent: doc?.pageContent,
+        metadata: { ...doc?.metadata, score },
```

Comment:
Can you reverse this so that if there's an explicit metadata field called `score` it won't be overridden?

`metadata: { score, ...doc?.metadata },`

I think it also might be better if there were a property like `returnScores` and the metadata were populated in `similaritySearchVectorWithScore` above.

Reply:
ok for the reverse.

`similaritySearchVectorWithScore` is already returning scores but with type `[Document<Record<string, any>>, number][]` . So for me would be redundant to add another flag for it. Is responsability of MMR method (since is using `similaritySearchVectorWithScore` method) to shape is own output.